src/simsoptpp/biot_savart_impl.h

```diff
@@ -284,3 +538,142 @@ void biot_savart_kernel_A(AlignedPaddedVec& pointsx, AlignedPaddedVec& pointsy,
         }
     }
 }
+
+#else
+
+template<class T, int derivs>
+void biot_savart_kernel_A(AlignedPaddedVec& pointsx, AlignedPaddedVec& pointsy, AlignedPaddedVec& pointsz,
+            T& gamma, T& dgamma_by_dphi, T& A, T& dA_by_dX, T& d2A_by_dXdX) {
+    if(gamma.layout() != xt::layout_type::row_major)
+          throw std::runtime_error("gamma needs to be in row-major storage order");
+    if(dgamma_by_dphi.layout() != xt::layout_type::row_major)
+          throw std::runtime_error("dgamma_by_dphi needs to be in row-major storage order");
+    int num_points         = pointsx.size();
+    int num_quad_points    = gamma.shape(0);
+    constexpr int simd_size = 1;
+    auto dA_dX_i = vector<Vec3dStd>();
+    if constexpr(derivs > 0) {
+        dA_dX_i = vector<Vec3dStd>{
+            Vec3dStd(), Vec3dStd(), Vec3dStd()
+        };
+    }
+    auto d2A_dXdX_i = vector<Vec3dStd>();
+    if constexpr(derivs > 1) {
+        d2A_dXdX_i = vector<Vec3dStd>{
+            Vec3dStd(), Vec3dStd(), Vec3dStd(),
+            Vec3dStd(), Vec3dStd(), Vec3dStd(),
+            Vec3dStd(), Vec3dStd(), Vec3dStd()
+        };
+    }
+    double fak = (1e-7/num_quad_points);
+    double* gamma_j_ptr = &(gamma(0, 0));
+    double* dgamma_j_by_dphi_ptr = &(dgamma_by_dphi(0, 0));
+    // out vectors pointsx, pointsy, and pointsz are added and aligned, so we
+    // don't have to worry about going out of bounds here
+    for(int i = 0; i < num_points; i += simd_size) {
+        auto point_i = Vec3dStd(&(pointsx[i]), &(pointsy[i]), &(pointsz[i]));
+        auto A_i   = Vec3dStd();
+        if constexpr(derivs > 0) {
+            dA_dX_i[0] *= 0.;
+            dA_dX_i[1] *= 0.;
+            dA_dX_i[2] *= 0.;
+        }
+        if constexpr(derivs > 1) {
+            d2A_dXdX_i[0] *= 0.; d2A_dXdX_i[1] *= 0.; d2A_dXdX_i[2] *= 0.;
+            d2A_dXdX_i[3] *= 0.; d2A_dXdX_i[4] *= 0.; d2A_dXdX_i[5] *= 0.;
+            d2A_dXdX_i[6] *= 0.; d2A_dXdX_i[7] *= 0.; d2A_dXdX_i[8] *= 0.;
+        }
+        for (int j = 0; j < num_quad_points; ++j) {
+            auto diff = point_i - Vec3dStd(gamma_j_ptr[3*j+0], gamma_j_ptr[3*j+1], gamma_j_ptr[3*j+2]);
+            auto norm_diff_2     = normsq(diff);
+            auto norm_diff_inv   = rsqrt(norm_diff_2);
+            auto norm_diff_3_inv = norm_diff_inv*norm_diff_inv*norm_diff_inv;
+
+            auto dgamma_by_dphi_j_simd = Vec3dStd(dgamma_j_by_dphi_ptr[3*j+0], dgamma_j_by_dphi_ptr[3*j+1], dgamma_j_by_dphi_ptr[3*j+2]);
+            /*
+            A_i.x += dgamma_by_dphi_j_simd.x * norm_diff_inv;
+            A_i.y += dgamma_by_dphi_j_simd.y * norm_diff_inv;
+            A_i.z += dgamma_by_dphi_j_simd.z * norm_diff_inv;
+            */
```

Comment:
Throughout the modified files there are some bits of commented-out code, such as here, which might be good to remove before merging.